expression/expr_to_pb.go

```diff
@@ -158,6 +159,14 @@ func (pc *PbConverter) encodeDatum(d types.Datum) (tipb.ExprType, []byte, bool)
 		if pc.client.IsRequestTypeSupported(kv.ReqTypeDAG, int64(tipb.ExprType_MysqlTime)) {
 			tp = tipb.ExprType_MysqlTime
 			t := d.GetMysqlTime()
+			// To be compatible with `encode` and `PBToExpr`, convert timestamp to UTC timezone.
```

Comment:
What about implement it as `EncodeMySQLTime` in `codec` package. And use it both here and `encode(sc *stmtctx.StatementContext, b []byte, vals []types.Datum, comparable bool, hash bool)` in `codec`.